lib/daps_functions

```diff
@@ -2113,33 +2124,58 @@ function spellcheck {
     # Set language - if not specified anywhere try to get from $MAIN
     #
     if [[ -n "$P_LANG" ]]; then
-        ASPELL_LANG="$P_LANG"
+        SPELL_LANG="$P_LANG"
     else
-        if [[ -z "$ASPELL_LANG" ]]; then
-            ASPELL_LANG=$($XSLTPROC --stylesheet "${DAPSROOT}/daps-xslt/common/get-language.xsl" --file "$MAIN" "$XSLTPROCESSOR" 2>/dev/null)
-            if [[ -n "$ASPELL_LANG" ]]; then
-        ccecho "info" "Using language $ASPELL_LANG."
+        if [[ -z "$SPELL_LANG" ]]; then
+            SPELL_LANG=$($XSLTPROC --stylesheet "${DAPSROOT}/daps-xslt/common/get-language.xsl" --file "$MAIN" "$XSLTPROCESSOR" 2>/dev/null)
+            if [[ -n "$SPELL_LANG" ]]; then
+                ccecho "info" "Using language $SPELL_LANG."
             else
-        exit_on_error "Failed to get language from $MAIN. Please specify it with --lang."
+                exit_on_error "Failed to get language from $MAIN. Please specify it with --lang."
             fi
         fi
     fi
 
     #------------------
-    # create spellchecking command
+    # create spellchecking command for both aspell and hunspell
     #
+
     # first, create skiplist
-    #
     COUNT=0
-    for SKIP in $ASPELL_SKIP_TAGS; do
-        SKIPLIST[$COUNT]="--add-sgml-skip=$SKIP"
+    for SKIP in $SPELL_SKIP_TAGS; do
+        case "$SPELL_CHECKER" in
+            aspell) SKIPLIST[$COUNT]="--add-sgml-skip=$SKIP"
+                ;;
+            hunspell) # hunspell supports no skipping
+                ;;
+        esac
         let COUNT++
     done
-    # now create the command
-    SPELLCMD="/usr/bin/aspell --mode=sgml ${SKIPLIST[*]} --encoding=utf-8 --lang=$ASPELL_LANG"
-    if [[ -n "$ASPELL_EXTRA_DICT" ]]; then
-        SPELLCMD="$SPELLCMD --extra-dicts=$ASPELL_EXTRA_DICT"
-    fi
+    # second, create the command
+    case "$SPELL_CHECKER" in
+        aspell )
+            SPELLCMD="$SPELL_CHECKER_CMD --mode=sgml ${SKIPLIST[*]} --encoding=utf-8 --lang=$SPELL_LANG"
+            if [[ -n "$SPELL_EXTRA_DICT" ]]; then
+                SPELLCMD+=" --extra-dicts=$SPELL_EXTRA_DICT"
+            fi
+            ;;
+        hunspell )
+            # expand dict name by lang
+            case "$SPELL_LANG" in
+                en)
+                    SPELL_LANG+="_US"
+                    ;;
+                cs)
+                    SPELL_LANG+="_CZ"
+                    ;;
+            esac
```

Comment:
Actually I think this part is questionable. We do not know which dictionary the user wants to use and therefore should not hardcode it here. If the user has not explicitly specified en_US with --lang, hunspell should use the generic "en" obtained from xml:lang.